assembly/src/assembler/mod.rs

```diff
@@ -286,6 +287,40 @@ impl Assembler {
         Ok(CompiledLibrary::new(mast_forest_builder.build(), exports)?)
     }
 
+    /// Assembles the provided module into a [CompiledLibrary] intended to be used as a Kernel.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if parsing or compilation of the specified modules fails.
+    pub fn assemble_kernel(mut self, module: impl Compile) -> Result<CompiledLibrary, Report> {
```

Comment:
We should reuse the implementation in `Assembler::assemble_library()` - the only difference between the 2 is the compile options. So we could move the implementation to a private helper function that both `assemble_{library, kernel}` use

Reply:
i tried to combine these but in the end decided to keep them separate for now. I think relatively soon we may discover that processing would different slightly between the two and we'd need to undo this work. But if it turns out that they look still pretty similar we can combine them later on.